poc/src/main.rs

```diff
@@ -140,6 +145,9 @@ async fn main() {
     if cli.memory_db {
         let db = akd::storage::memory::AsyncInMemoryDatabase::new();
         let mut directory = Directory::<_>::new::<Blake3>(&db).await.unwrap();
+        if let Some(()) = pre_process_input(&cli, &tx, None).await {
```

Comment:
Is this the right position? This is specifically when a user has selected "in-memory database"

Reply:
Ah the idea was to have both methods call the same pre-processing function. At this time there's only logic relevant to mysql, but in future we could have some logic shared between both or even memory specific logic. Happy to remove for now if that's what we prefer!